### PR TITLE
feat: update to support PHP 8

### PR DESCRIPTION
**Pull request by owenvoke**
It looks like the following also need to be updated to support PHP 8 upstream.

- ~~[`laminas/laminas-servicemanager`](https://github.com/laminas/laminas-servicemanager)~~
- ~~[`ramsey/uuid`](https://github.com/ramsey/uuid) (looks like it now supports PHP 8 in master, but waiting on a release)~~ (It's had support for ages, I just misread the repository)
- ~~[`nunomaduro/laravel-console-dusk`](https://github.com/nunomaduro/laravel-console-dusk) (waiting on new [Laravel Dusk](https://github.com/laravel/dusk) and [PHP Webdriver](https://github.com/php-webdriver/php-webdriver/pull/828) releases)~~
- ~~[`hmazter/laravel-schedule-list`](https://github.com/hmazter/laravel-schedule-list) (waiting on a new tag, support is available in `master`)~~
- ~~The packaged Box binary under `bin/` (https://github.com/box-project/box/issues/486) once they create a new PHAR release~~

We will also want to update the Updater component to use the new [Laravel Zero fork of PHAR Updater](https://github.com/laravel-zero/phar-updater).

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "require": {
         "php": "^7.3",
```
Comment (spresnac-trox) [inline composer.json:18]: That should be `^7.3 | ^8` otherwise, you cannot install zero on a system with php8 only 😉

**Comment by nunomaduro** [comment]
cc @owenvoke

File: composer.json (hunk 1 above)
Comment (owenvoke) [reply composer.json:18]: I'm not sure why I was thinking of updating this at the end. 🤦🏻

**Comment by owenvoke** [comment]
@hmazter, is there any chance you could tag a release of [Laravel Schedule List](https://github.com/hmazter/laravel-schedule-list) with PHP 8 support? 🙂

**Comment by hmazter** [comment]
> @hmazter, is there any chance you could tag a release of [Laravel Schedule List](https://github.com/hmazter/laravel-schedule-list) with PHP 8 support? 🙂

Sorry, thought that I already had. Will fix it in 2 hours. :smile:

**Comment by owenvoke** [comment]
No problem, thank you!

**Comment by hmazter** [comment]
> > @hmazter, is there any chance you could tag a release of [Laravel Schedule List](https://github.com/hmazter/laravel-schedule-list) with PHP 8 support? 🙂
> 
> Sorry, thought that I already had. Will fix it in 2 hours. 😄

v2.2.1 tagged https://packagist.org/packages/hmazter/laravel-schedule-list#v2.2.1

**Comment by nunomaduro** [comment]
@owenvoke Let me know If I can help here.

**Comment by owenvoke** [comment]
I think Box is the only thing we are waiting on now. :+1: I'm going to have a check through the internal framework code to check that's all compatible as well.

**Comment by nunomaduro** [comment]
We should be good on that regard, as we have some tests. And if something is not good, the community can report and we patch it. So we need to wait for box now...

**Comment by owenvoke** [comment]
Right, Box v3.11.0 is now released which adds PHP 8 support. I will add this in and try to aim for a release soon. :+1:

**Review by nunomaduro — APPROVED** [review]

**Comment by bhamail** [comment]
Is this perhaps another dependency that needs to be updated for php8?:

[intonate/tinker-zero]()

see: ["php": "^7.2.5"](https://github.com/intonate/tinker-zero/blob/master/composer.json#L18)

With a dependency on `"laravel-zero/framework": "^8.0"` (and `"php": "^8.0",`), I get the following `composer update` error:

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - intonate/tinker-zero[dev-master, v1.1.0] require php ^7.2.5 -> your php version (8.0.0) does not satisfy that requirement.
    - Root composer.json requires intonate/tinker-zero >1.0 -> satisfiable by intonate/tinker-zero[v1.1.0].
```

**Comment by owenvoke** [comment]
@bhamail, Tinker Zero isn't an official package so they'd have to add support over there. Feel free to open a PR on their repository, or I'll PR it tomorrow.

**Comment by bhamail** [comment]
@owenvoke Cool. Thanks! 

Just in case others hit a similar issue, in my case, we don't need `tinker-zero` in php8, so removing it was fine, until I forgot to remove the `providers` entry. See: https://github.com/sonatype-nexus-community/bach/pull/21/commits/f491d71ad2f3103206349ec16cc436a13ea4f83d

**Comment by owenvoke** [comment]
I've opened a PR over at https://github.com/intonate/tinker-zero/pull/8, so hopefully that can get merged. 👍🏻

Anyway, glad you got it sorted.

**Pull request merged**